translator/translate/otel/processor/awsappsignals/translator.go

```diff
@@ -66,9 +66,22 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 		if !hostedInConfigured {
 			hostedIn = util.GetClusterNameFromEc2Tagger()
 		}
-		cfg.Resolvers = []appsignalsconfig.Resolver{
-			appsignalsconfig.NewEKSResolver(hostedIn),
+
+		isEks, err := common.IsEKS()
+		if err != nil {
+			return nil, err
 		}
+
+		if isEks {
+			cfg.Resolvers = []appsignalsconfig.Resolver{
+				appsignalsconfig.NewEKSResolver(hostedIn),
+			}
+		} else {
+			cfg.Resolvers = []appsignalsconfig.Resolver{
+				appsignalsconfig.NewK8sResolver(hostedIn),
```

Comment:
The logic within the if else blocks are all different. The common code would be lines 70 to 73, but that can't be exported because we would still need to return if an error occurs.